pkg/kv/kvprober/kvprober.go

```diff
@@ -335,9 +368,21 @@ func (p *Prober) Start(ctx context.Context, stopper *stop.Stopper) error {
 					return
 				}
 
-				probeCtx, sp := tracing.EnsureChildSpan(ctx, p.tracer, opName+" - probe")
-				probe(probeCtx, pl)
-				sp.Finish()
+				func() {
+					// trace kv prober requests
+					probeCtx, collectAndFinish := tracing.ContextWithRecordingSpan(ctx, p.tracer, opName)
+
+					defer collectAndFinish()
+					if probe(probeCtx, pl) {
+						recording := collectAndFinish()
+
+						if traceDumpLimiter.ShouldLog() {
+							log.Health.Infof(ctx, "%v", recording)
```

Comment:
@arulajmani, do you have an opinion on what channel we should dump the trace to? We could introduce a new channel just for these traces.